go-controller/cmd/ovn-k8s-overlay/app/minion_init.go

```diff
@@ -0,0 +1,120 @@
+package app
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/openvswitch/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/urfave/cli"
+)
+
+// InitMinionCmd initializes k8s minion node
+var InitMinionCmd = cli.Command{
+	Name:  "minion-init",
+	Usage: "Initialize k8s minion node",
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "cluster-ip-subnet",
+			Usage: "The cluster wide larger subnet of private ip addresses.",
+		},
+		cli.StringFlag{
+			Name:  "minion-switch-subnet",
+			Usage: "The smaller subnet just for this master.",
+		},
+		cli.StringFlag{
+			Name:  "node-name",
+			Usage: "A unique node name.",
+		},
+		cli.StringFlag{
+			Name:  "nb-privkey",
+			Usage: "The private key used for northbound API SSL connections.",
+		},
+		cli.StringFlag{
+			Name:  "nb-cert",
+			Usage: "The certificate used for northbound API SSL connections.",
+		},
+		cli.StringFlag{
+			Name:  "nb-cacert",
+			Usage: "The CA certificate used for northbound API SSL connections.",
+		},
+	},
+	Action: func(context *cli.Context) error {
+		if err := initMinion(context); err != nil {
+			return fmt.Errorf("failed init minion: %v", err)
+		}
+		return nil
+	},
+}
+
+func initMinion(context *cli.Context) error {
+	err := fetchOVNNB(context)
+	if err != nil {
+		return err
+	}
+	minionSwitchSubnet := context.String("minion-switch-subnet")
+	if minionSwitchSubnet == "" {
+		return fmt.Errorf("argument --minion-switch-subnet should be non-null")
+	}
+
+	clusterIPSubnet := context.String("cluster-ip-subnet")
+	if clusterIPSubnet == "" {
+		return fmt.Errorf("argument --cluster-ip-subnet should be non-null")
+	}
+
+	nodeName := context.String("node-name")
+	if nodeName == "" {
+		return fmt.Errorf("argument --node-name should be non-null")
+	}
+
+	cniPluginPath, err := exec.LookPath(config.CniPlugin)
```

Comment:
The lines from here for CNI configuration are only run for Linux (skipped for windows). Can you just do the same  here too.

Reply:
updated